include/dslash_helper.cuh

```diff
@@ -722,19 +730,42 @@ namespace quda
         const int dslash_block_offset
           = ((kernel_type == INTERIOR_KERNEL || kernel_type == UBER_KERNEL) ? arg.pack_blocks : 0);
         int x_cb = (target::block_idx().x - dslash_block_offset) * target::block_dim().x + target::thread_idx().x;
-        if (x_cb >= arg.threads) return;
 
-#ifdef QUDA_FAST_COMPILE_DSLASH
-        dslash.template operator()<kernel_type == UBER_KERNEL ? INTERIOR_KERNEL : kernel_type>(x_cb, s, parity);
+#ifdef NVSHMEM_COMMS
+        constexpr bool use_nvshmem_comms = true;
 #else
-        switch (parity) {
-        case 0: dslash.template operator()<kernel_type == UBER_KERNEL ? INTERIOR_KERNEL : kernel_type>(x_cb, s, 0); break;
-        case 1: dslash.template operator()<kernel_type == UBER_KERNEL ? INTERIOR_KERNEL : kernel_type>(x_cb, s, 1); break;
-        }
+        constexpr bool use_nvshmem_comms = false;
 #endif
+        if constexpr (use_nvshmem_comms && Arg::D::use_syncthreads) {
 #ifdef NVSHMEM_COMMS
```

Comment:
Is this #ifdef guard needed here?  I think the code will be hidden from the compiler anyway since it's in the `if constexpr (use_nvshmem_comms....)`

Reply:
Somehow it does not - it complains about not finding `cuda::atomic`.